docs/sources/setup-grafana/upgrade-grafana.md

```diff
@@ -405,6 +405,91 @@ This behavior was not very intuitive and creates issues for users who want to ch
 That is why we are changing the behavior in 8.5. From now on, the `default` data source will not be a persisted property but just the starting data source for new panels and queries.
 Existing dashboards that still have panels with a `datasource` set to null will be migrated when the dashboard opens. The migration will set the data source property to the **current** default data source.
 
+## Upgrading to 9.0
+
+### Role-based access control: changes for general release
+
+Fine-grained access control is now called "Role-based access control (RBAC)". As part of the Grafana 9.0 release, the feature is generally available, and there are several breaking changes:
+
+- Built-in roles are now called basic roles. They now consist of permissions, not roles.
+- The Terraform `builtin_role_assignment` resource is deprecated. Please use [grafana_role](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/role) resource instead.
```

Comment:
This is actually not a very smooth experience, I've documented the steps that you need to take here: https://github.com/grafana/grafana-enterprise/issues/4108

I think we should also include that somewhere in the docs (not as part of this PR obviously, just wanted to call it out).